cardano-sync/src/Cardano/Sync/LedgerState.hs

```diff
@@ -128,7 +127,7 @@ data LedgerEnv = LedgerEnv
   { leProtocolInfo :: !(Consensus.ProtocolInfo IO CardanoBlock)
   , leDir :: !LedgerStateDir
   , leNetwork :: !Ledger.Network
-  , leStateVar :: !(StrictTVar IO CardanoLedgerState)
+  , leStateVar :: !(StrictTVar IO (Maybe CardanoLedgerState))
```

Comment:
I wonder if we could have an empty or minimal ledger state there instead of having the `Maybe` .

Reply:
We read `leStateVar` in 1 places: `applyBlock`. If we have a minimal state at this point it would simply lead to more cryptic errors.

We could try to make this type safe. This would probably need to follow the approach of ouroboros-network with typed protocols and parametrise `LedgerEnv` over the `Nat n`. Probably a pretty big refactoring.

`leStateVar` can be empty in 2 cases: after initiation and if loadLedgerAtPoint returns Left. In both cases we send to the node a `FindIntersect` message. The node replies with the point we should roll back and so `loadLedgerAtPoint` is called again.